src/v/model/transform.cc

```diff
@@ -188,6 +188,7 @@ model::record_batch transformed_data::make_batch(
     // transforms.
     header.producer_id = -1;
 
+    header.last_offset_delta = i - 1;
```

Comment:
why -1 ?

Reply:
Good question:

the delta is used to compute `last_offset`. If you have a batch with offsets 4, 5, 6. Then the delta is 2, because base offset is 4, last offset is 4 + 2 = 6. 

storage batch builder does the same thing here:

https://github.com/redpanda-data/redpanda/blob/9d8493cfce44d1265dc096dc0bd665dbb93d9433/src/v/storage/record_batch_builder.cc#L70

Reply:
thanks !